src/compiler/checker.ts

```diff
@@ -21573,6 +21573,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             const t = isFreshLiteralType(type) ? (type as FreshableType).regularType :
                 isGenericTupleType(type) ? getNormalizedTupleType(type, writing) :
                 getObjectFlags(type) & ObjectFlags.Reference ? (type as TypeReference).node ? createTypeReference((type as TypeReference).target, getTypeArguments(type as TypeReference)) : getSingleBaseForNonAugmentingSubtype(type) || type :
+                type.flags & TypeFlags.EnumLiteral && type.flags & TypeFlags.Union ? getUnionType((type as UnionType).types) :

```

Comment:
Not sure if I should cache the created type, e.g. `type.resolvedReducedType || type.resolvedReducedType = getUnionType((type as UnionType).types)`.

Note that this must go above the next line, since `getNormalizedUnionOrIntersectionType` will just return back `type`, which means the current loop will terminate.